config/7.1.0/securebanking/ig/scripts/groovy/ConsentIdValidator.groovy

```diff
@@ -0,0 +1,34 @@
+import org.forgerock.http.protocol.*
+import com.forgerock.sapi.gateway.rest.HttpHeaderNames
+import groovy.json.JsonSlurper
+
+/**
+ * This script is comparing the consentId from the request payload versus the consentId from the provided access token.
+ * By doing this comparison, we prevent resources being submitted with a token obtained for another consent.
+ */
+
+String fapiInteractionId = request.getHeaders().getFirst(HttpHeaderNames.X_FAPI_INTERACTION_ID);
+if (fapiInteractionId == null) { fapiInteractionId = 'No ' + HttpHeaderNames.X_FAPI_INTERACTION_ID + ' header'}
+SCRIPT_NAME = '[ConsentIdValidator] (' + fapiInteractionId + ') - '
+logger.debug(SCRIPT_NAME + 'Running...')
+
+// Get the intent id from the access token
+def accessTokenIntentId = attributes.openbanking_intent_id
```

Comment:
Can you add a check for this attribute being missing please? So that we can fail fast here.

If the attribute is missing, then we can assume that `SaveIntentIdOnAttributesContext` is missing or miss configured in the filter chain. This is therefore an application configuration issue.

Log a suitable error and throw an IllegalStateException (IG will then return a HTTP 500). This is how we handle such issues in the java filters.

Reply:
Added the validation for the attribute. The behaviour is as follows:

An IllegalStateException error is thrown which result in this error exception in the logs:
`java.lang.IllegalStateException: openbanking_intent_id claim is missing from the access token`
and the caller will get this error response:

```json
{
    "Code": "500",
    "Id": "3b22668cdb2ab48ce91a1090079af85b",
    "Message": "[Status: 500 Internal Server Error]",
    "Errors": [
        {
            "ErrorCode": "UK.OBIE.UnexpectedError",
            "Message": "Internal error"
        }
    ]
}
```

Reply:
Good point, updated the error message. Also added an error in the SaveIntentIdOnAttributesContext script.